documentation/optee_design.md

```diff
@@ -777,6 +778,66 @@ consists of:
 | `uint8_t[hash_size]` | hash | Hash of the fields above and the `<ELF>` above |
 | `uint8_t[sig_size]` | signature | Signature of hash |
 
+# 13. Device Tree support
+
+OP-TEE core can use the device tree format to inject platform configuration
+information during platform initialization and possibly some run time context.
+
+Device Tree technology allows to describe platforms from ASCII source files
+so-called DTS files. These can be used to generate a platform description
+binary image embedded in the platform boot media for applying expected
+configuration settings during the platform initializations.
+
+This scheme relaxes design constrains on the OP-TEE core implementation as
+most of the platform specific hardware can be tuned without modifying C
+source files or adding configuration directives in the build environments.
+
+## Secure and Non Secure Device Trees
+
+When OP-TEE is embedded into a platform, it is likely there exist a non secure
+world which has some platform hardware to handle. A non secure kernel as the
+Linux kernel can use device tree files to describe the hardware.
+
+OP-TEE can embed a device tree image for initializing platform drivers.
+
+Maybe the early bootloader loaded a device tree image, or several, for the
+benefit of OP-TEE secure OS and/or the non secure world.
+
+As one can see, there can be several device trees and some can be shared
```

Comment:
I would have skipped "As one can see, " and just started the sentence with "There..."

Reply:
ok, thanks.